misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JDIEventProcessor.java

```diff
@@ -144,6 +166,20 @@ private boolean processEvent(EventSet eventSet, Event event) {
         return true;
     }
 
+    private boolean evaluateBreakpointCondition(String expression, ThreadReference threadReference) {
+        try {
+            StackFrameProxyImpl frame = context.getAdapter().getAllThreads().get(threadReference.uniqueID()).frame(0);
+            SuspendedContext ctx = new SuspendedContext(context.getSourceProject(), context.getDebuggeeVM(),
+                    context.getAdapter().getAllThreads().get(threadReference.uniqueID()), frame);
+            ExpressionEvaluator evaluator = new ExpressionEvaluator(ctx);
+            String condition = evaluator.evaluate(expression).toString();
+            return condition.equalsIgnoreCase(CONDITION_TRUE);
+        } catch (JdiProxyException e) {
+            LOGGER.error(e.getMessage(), e);
+            return false;
+        }
+    }
```

Comment:
In here, Your logic returns `false` for the expressions which produces a non-boolean results as well. Shall we improve this behavior by notifying the user that it is an invalid expression instead?

Reply:
Yes, we can notify the user when they add an invalid expression. 
I have created https://github.com/ballerina-platform/ballerina-lang/issues/28992 to track this.